packages/api/src/derives/attestation/attestationRx.e2e.ts

```diff
@@ -0,0 +1,261 @@
+
+      claim
+        .signAndSend(issuer)
+        .pipe(
+          filter(({ events, status }) => {
+            return status.isInBlock && events !== undefined;
+          })
+        )
+        .subscribe(({ events, status }) => {
+          for (const {
+            event: { method, data },
+          } of events) {
+            if (method === 'ClaimUpdated') {
+              expect(data[0].toString()).toBe(holder.address);
+              // Expect issuers to match
+              expect(data[1].toString()).toBe(issuer.address);
+              // expect topic to match
+              expect(data[2].toString()).toEqual(topic2);
+              expect(data[3].toString()).toBe(attestationValue2.toHex());
+              done();
+            }
+          }
+        });
+    });
+
+    it('should create a claim with issuer 2 and topic 1', done => {
+      const claim = api.tx.attestation.setClaim(holder.address, topic, attestationValue.toHex());
+
+      claim
+        .signAndSend(issuer2)
+        .pipe(
+          filter(({ events, status }) => {
+            return status.isInBlock && events !== undefined;
+          })
+        )
+        .subscribe(({ events, status }) => {
+          for (const {
+            event: { method, data },
+          } of events) {
+            if (method === 'ClaimUpdated') {
+              expect(data[0].toString()).toBe(holder.address);
+              // Expect issuers to match
+              expect(data[1].toString()).toBe(issuer2.address);
+              // expect topic to match
+              expect(data[2].toString()).toEqual(topic);
+              expect(data[3].toString()).toBe(attestationValue.toHex());
+              done();
+            }
+          }
+        });
+    });
+
+    it('should create a claim with issuer 2 and topic 2', done => {
+      const claim = api.tx.attestation.setClaim(holder.address, topic2, attestationValue2.toHex());
+
+      claim
+        .signAndSend(issuer2)
+        .pipe(
+          filter(({ events, status }) => {
+            return status.isInBlock && events !== undefined;
+          })
+        )
+        .subscribe(({ events, status }) => {
+          for (const {
+            event: { method, data },
+          } of events) {
+            if (method === 'ClaimUpdated') {
+              expect(data[0].toString()).toBe(holder.address);
+              // Expect issuers to match
+              expect(data[1].toString()).toBe(issuer2.address);
+              // expect topic to match
+              expect(data[2].toString()).toEqual(topic2);
+              expect(data[3].toString()).toBe(attestationValue2.toHex());
+              done();
+            }
+          }
+        });
+    });
+  });
+
+  describe('Get Mutiple Claims Claim', () => {
+    it('should get a claim with a specific issuer and holder', done => {
+      api.derive.attestation
+        .getClaims(holder.address, [issuer.address, issuer2.address], [topic, topic2])
```

Comment:
Isn't it possible that this test does nothing and still passes (false positive) i.e. if doesn't go into the `if` blocks?
I wonder if I may've broken the API here as it used to pass ok I guess?
`.subscribe(({claims}) => {`
either way this test needs to be reworked

Reply:
This can happen sometime when the extrinsic is not yet applied and subscribe would give value before and also once extrinsic is submitted and storage is changed.. I have updated the test..